pkg/cluster/cluster_config.go

```diff
@@ -21,6 +22,69 @@ import (
 
 // +kubebuilder:rbac:groups="config.openshift.io",resources=ingresses,verbs=get
 
+type Platform string
+
+// Release includes information on operator version and platform
+// +kubebuilder:object:generate=true
+type Release struct {
+	Name    Platform                `json:"name,omitempty"`
+	Version version.OperatorVersion `json:"version,omitempty"`
+}
+
+var clusterConfig struct {
+	Namespace string
+	Platform  Platform
+	Release   Release
+}
+
+// Init initializes cluster configuration variables on startup
+// init() won't work since it is needed to check the error.
+func Init(ctx context.Context, cli client.Client, log logr.Logger) error {
+	var err error
+
+	clusterConfig.Namespace, err = getOperatorNamespace()
+	if err != nil {
+		log.Error(err, "unable to find operator namespace")
+		// not fatal, fallback to ""
+	}
+
+	clusterConfig.Platform, err = getPlatform(ctx, cli)
+	if err != nil {
+		return err
+	}
+
+	clusterConfig.Release, err = getRelease(ctx, cli)
+	if err != nil {
+		return err
+	}
+
+	printClusterConfig(log)
+
+	return nil
+}
+
+func printClusterConfig(log logr.Logger) {
+	log.Info("Cluster config",
+		"Namespace", clusterConfig.Namespace,
+		"Platform", clusterConfig.Platform,
+		"Release", clusterConfig.Release)
+}
+
+func GetOperatorNamespace() (string, error) {
```

Comment:
so 
`getOperatorNamespace()` call with real logic and return retrieve value
`GetOperatorNamespace(...) `return clusterConfig.Namespace


similar to 
`getPlatform() ` and `getRelease()` call logic return value
`GetPlatform(...)` and `GetRelease(...)` return filed's value

GetRelease(...) indirectly rely on `getPlatform() `